src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -123,15 +125,8 @@ program::program(engine& engine_ref,
     }
 }
 
-program::program(engine& engine_ref,
-                 std::set<std::shared_ptr<program_node>> const& nodes,
-                 build_options const& options,
-                 bool is_internal)
-    : _engine(engine_ref),
-      _stream(_engine.create_stream()),
-      options(options),
-      processing_order(),
-      tuning_cache(nullptr) {
+void program::init_program(std::set<std::shared_ptr<program_node>> const& nodes,
+                           bool is_internal) {
```

Comment:
BTW why shouild we separate this from the constructor?

Reply:
In order to obtain the shared_ptr of the `program`, make_shared must be returned. In the current design, all `program_nodes` are constructed in the construtor of `program`. It means that it's not possible to get the shared_ptr of `program` when `program_nodes` are constructed. So, I divided the constructor of `program` into two parts.